src/main/java/org/ow2/proactive/catalog/service/BucketService.java

```diff
@@ -84,32 +84,50 @@ public BucketMetadata getBucketMetadata(String bucketName) {
         return new BucketMetadata(bucketEntity);
     }
 
-    public List<BucketMetadata> listBuckets(List<String> owners, String kind) {
+    public List<BucketMetadata> listBuckets(List<String> owners, String kind, String contentType) {
+        System.out.println("params kind  " + kind + " content " + contentType);
         if (owners == null) {
             return Collections.emptyList();
         }
 
         List<BucketEntity> entities;
-        if (!StringUtils.isEmpty(kind)) {
+        if (!StringUtils.isEmpty(kind) && !StringUtils.isEmpty(contentType)) {
+            System.out.println("kind  " + kind + " content " + contentType);
+            entities = bucketRepository.findByOwnerIsInContainingKindAndContentType(owners, kind, contentType);
+        } else if (!StringUtils.isEmpty(kind) && StringUtils.isEmpty(contentType)) {
+            System.out.println("kind  " + kind);
             entities = bucketRepository.findByOwnerIsInContainingKind(owners, kind);
-        } else {
+        } else if (StringUtils.isEmpty(kind) && !StringUtils.isEmpty(contentType)) {
+            System.out.println("content " + contentType);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)